src/credentials_callback.cc

```diff
@@ -0,0 +1,38 @@
+/**
+ * \file   credentials_callback.cc
+ * \author Lars Fröhlich
+ * \date   Created on January 16, 2024
+ * \brief  Implementation of the credentials callback function.
+ *
+ * \copyright Copyright 2024 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include <git2.h>
+
+namespace git {
+extern "C" {
+
+int credentials_callback(git_cred** out_credentials, const char* /*url*/,
+    const char* /*username_from_url*/, unsigned int /*allowed_types*/, void* /*payload*/)
+{
+    // Dummy implementation: Always return the same username and password
+    return git_cred_userpass_plaintext_new(out_credentials, "user", "pwd");
```

Comment:
:thinking: 

4 years ago they changed `cred` to `credential`.
The API documentation lists (only) the long forms.

![image](https://github.com/taskolib/libgit4cpp/assets/16012374/7f9c0b7f-e639-48e5-854b-15329352b9d9)

Here the commit over there:

https://github.com/libgit2/libgit2/commit/3f54ba8b61869f42b2bbd1a60091a0be640bc8fc

## Package situation

Ubuntu 20.04 has the ancient libgit, while Ubuntu 22.04 has libgit 1.1.0 - **so this PR will not build on Ubuntu 22.04**

![image](https://github.com/taskolib/libgit4cpp/assets/16012374/d71068c2-9152-4d30-990d-0d4667ffac01)

I'm not sure why it builds in the CI... compare `doocsdev22` below

![image](https://github.com/taskolib/libgit4cpp/assets/16012374/504f00b5-aeb8-48cd-83af-6a4f9612d785)

![image](https://github.com/taskolib/libgit4cpp/assets/16012374/0fc04988-24e9-4bf7-8762-65ffaf3a80bc)

Reply:
Hmm, I remember that I had some `meson.build` that checked the `gitlib2` version and
* Output "Ancient libgit2 version found" (or similar)
* Set some macro that code could use in `#ifdef`
* Used that macro for some library function that changed behavior with v1.0.0

But I can not find it and do not see it anymore :thinking:

Reply:
Ah, here:

```python
## libgit2 has obviously a change in the globbing
local_compiler_args = []
if dependency('libgit2').version().version_compare('<1')
    message('Ancient libgit2 version detected')
    local_compiler_args += [ '-DANCIENT_LIBGIT2' ]
endif
```

The comparison is obviously wrong because the change happened already with 0.99.0, but that is not released as package because it is a preview version.

https://github.com/taskolib/taskolib/commit/afb384be8173e5b4e1a5338df68af8cbb32c49df

And ... why does `taskolib` need to know about the libgit details, that should be encapsulated here?

Reply:
>  I am going to remove that part from Taskolib.

Actually I cannot remove it because the test for ANCIENT_LIBGIT2 is used inside Taskolib to determine if a glob should end in "/*" or not. So if the test is needed in Taskolib, it should also be done in Taskolib.

Reply:
Well I meant ... should not `libgit4cpp` represent one globbing way, and IT adapts the glob to ancient libgit if needed?
So the user does not nee to know that the globbing might change with underlying version.

This is just an idea. The macro is kind of suboptimal, because one needs to know that it is needed in your own code.
Maybe I will try to come up with a PR that unifies the globbing stuff; but I guess before that I would need to examine when it has been introduced into libgit and why and what else etc pp. ;-)